README.md

```diff
@@ -55,7 +55,13 @@ Answer 'Yes' when prompted:
 Do you want to use staking in this service? (yes/no): y
 ```
 
-__Notes__: 
+Find below a diagram of the possible status a service can be in the **Alpine staking** program:
+
+![Alpine staking FSM](images/alpine_staking_fsm.svg)
+
+Services can become staked by invoking the `stake()` contract method, where service parameters and deposit amounts are verified. Staked services can call the `checkpoint()` method at regular intervals, ensuring liveness checks and calculating staking incentives. In case a service remains inactive beyond the specified `maxAllowedInactivity` time, it faces eviction from the staking program, ceasing to accrue additional rewards. Staked or evicted services can unstaked by calling the `unstake()` contract method. They can do so after `minStakingDuration` has passed.
```

Comment:
```suggestion
Services can become staked by invoking the `stake()` contract method, where service parameters and deposit amounts are verified. Staked services can call the `checkpoint()` method at regular intervals, ensuring liveness checks and calculating staking incentives. In case a service remains inactive beyond the specified `maxAllowedInactivity` time, it faces eviction from the staking program, ceasing to accrue additional rewards. Staked or evicted services can unstaked by calling the `unstake()` contract method. They can do so after `minStakingDuration` has passed or if there are no staking more reward available. 
```